### PR TITLE
Use common currency value object

### PR DESCRIPTION
**Pull request by benjefferies**
The OBIE spec has chosen to create a new currency value object
for storing the exact same values. For clean code purposes we've
refactored this.

### CODE REVIEW

File: src/main/java/uk/org/openbanking/datamodel/payment/OBDomestic1.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
   private String localInstrument = null;
 
   @JsonProperty("InstructedAmount")
-  private OBDomestic1InstructedAmount instructedAmount = null;
```
Comment (qcastel) [inline src/main/java/uk/org/openbanking/datamodel/payment/OBDomestic1.java]: Can you not delete OBDomestic1InstructedAmount? (same comment for the other classes)

Comment (benjefferies) [reply src/main/java/uk/org/openbanking/datamodel/payment/OBDomestic1.java]: I have deleted it https://github.com/ForgeRock/OpenBanking-Java-SDK/pull/21/files#diff-eb4fee6d34133203c7ea2d85a68ac834

Comment (qcastel) [reply src/main/java/uk/org/openbanking/datamodel/payment/OBDomestic1.java]: ah yea, my bad. not use to github review yet

**Comment by qcastel** [comment]
Let's also clean the other classes as well.

I can see there are:
- OBDomesticStandingOrder1FirstPaymentAmount
- OBDomesticStandingOrder1RecurringPaymentAmount
etc.

(searching for `@JsonProperty("Currency")` is a way to get the list of those duplicated objects)

**Comment by benjefferies** [comment]
Done @qcastel

**Pull request merged**